### ISSUE TITLE
Can't put null into the database?

### ISSUE THREAD

**Issue by MiralDesai** [satyan/sugar#603]
Why can't we put null into our database?

Some api calls are not going to return all data right away, but the column still needs to be created, and ideally nulls should be interested into the tables.

In SQLite I can do with putNull(key);

I'm currently looking into using SugarORM, and inserting data via from json returned from api calls. It's extremely common for API calls to return null for certain fields, so I would like to insert those nulls into the database.

Considering it's already supported in SQLite I'm surprised it's not available in Sugar.

**Comment by larrytech7** [comment]
I don't think you can use null for @unique fields. I think all the fields of your persistent class are used as columns and so when you don't set a value for a field, it is set as null. So when you retrieve the data, that filed will be null for the given instance. What have you tried so far?

**Comment by MiralDesai** [comment]
I have moved away from SugarOrm in favour of Realm. However when I was working on this I had helper methods that took the data from the json, whether it was null or not and inserted it into my database. So everything from the json was always updated into the database.

With SugarORM this through an exception. I guess it's not an issue for me anymore, but for other users of SugarOrm i would think this should be working.

I was essentially going sugarOrmoBJ.setId(null) for some data which caused an NPE exception.

**Comment by julianfalcionelli** [comment]
Any news about this ? As you guys says, i can't save null values. For example, i have a Class User which has an Address object, but sometimes the address is null and SugarORM don't allows me to save null in the Address column for the User Table.

**Comment by julianfalcionelli** [comment]
Well for the moment i will pass to Realm like @MiralDesai
